torchgeo/datasets/sentinel.py

```diff
@@ -25,6 +25,221 @@ class Sentinel(RasterDataset):
+      <https://asf.alaska.edu/>`_
+    * `Microsoft's Planetary Computer
+      <https://planetarycomputer.microsoft.com/dataset/sentinel-1-rtc>`_
+
+    Product Types:
+
+    * `Level-0
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-0>`_:
+      Raw (RAW)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Single Look Complex (SLC)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Ground Range Detected (GRD)
+    * `Level-2
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-2>`_:
+      Ocean (OCN)
+
+    Polarizations:
+
+    * HH: horizontal transmit, horizontal receive
+    * HV: horizontal transmit, vertical receive
+    * VV: vertical transmit, vertical receive
+    * VH: vertical transmit, horizontal receive
+
+    Acquisition Modes:
+
+    * `Stripmap (SM)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/stripmap>`_
+    * `Interferometric Wide (IW) swath
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/interferometric-wide-swath>`_
+    * `Extra Wide (EW) swatch
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/extra-wide-swath>`_
+    * `Wave (WV)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/wave>`_
+
+    .. note::
+       At the moment, this dataset only supports the GRD product type. Data must be
+       radiometrically terrain corrected (RTC). This can be done manually using a DEM,
+       or you can download an On Demand RTC product from ASF DAAC.
+
+    .. versionadded:: 0.4
+    """
+
+    # SAFE format
+    # https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/naming-conventions
+    # https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/document-library/-/asset_publisher/1dO7RF5fJMbd/content/sentinel-1-product-specification
+
+    # ASF DAAC GRD RTC uses a different naming scheme (README.md.txt in download):
+    #
+    # S1x_yy_aaaaaaaaTbbbbbb_ppo_RTCzz_u_defklm_ssss
+    #
+    # x:          Sentinel-1 Mission (A or B)
+    # yy:         Beam Mode
+    # aaaaaaaa:   Start Date of Acquisition (YYYYMMDD)
+    # bbbbbb:     Start Time of Acquisition (HHMMSS)
+    # pp:         Polarization: Dual-pol (D) vs. Single-pol (S),
+    #                 primary polarization (H vs. V)
+    # o:          Orbit Type: Precise (P), Restituted (R), or Original Predicted (O)
+    # zz:         Terrain Correction Pixel Spacing
+    # u:          Software Package Used: GAMMA (G)
+    # d:          Gamma-0 (g) or Sigma-0 (s) Output
```

Comment:
If you must put conditions on what values one dataset can have, then ensure that one instantiation can only have either $\gamma_0$ or $\sigma_0$. Otherwise differing images will mean vastly different things!

Reply:
We currently don't enforce this, but we could. But that would mean creating separate `Sentinel1Gamma` and `Sentinel1Sigma` classes. How common is it for a user to have a directory containing a mix of both gamma and sigma files?

Reply:
Or it could be a parameter of `__init__` and we would only need a single `Sentinel1` class. Would users be confused if they don't know what this thing means and TorchGeo can't find any data for them because they didn't specify the right option?

Reply:
It's not common to mix gamma and sigma. Let's resolve this comment as we can leave this up to the user. Can we add a comment noting that, though?